nwbinspector/nwbinspector.py

```diff
@@ -447,6 +447,13 @@ def inspect_nwb(
         Skip the PyNWB validation step. This may be desired for older NWBFiles (< schema version v2.10).
         The default is False, which is also recommended.
     """
+    # Check importance_threshold passed as string
+    if isinstance(importance_threshold, str):
+        if importance_threshold in Importance.__members__:
+            importance_threshold = getattr(Importance, importance_threshold)
+        else:
+            raise ValueError(f"'{importance_threshold}' is not a valid importance_threshold value.")
+
```

Comment:
This is a bit more complicated than we actually need for this: [note how the CLI handles it](https://github.com/NeurodataWithoutBorders/nwbinspector/blob/dev/nwbinspector/nwbinspector.py#L252) (`Enum` objects like our `Importance` are indexable just like dictionaries)

```python
importance_threshold = Importance[importance_threshold] if isinstance(importance_threshold, str) else importance_threshold
```

should be sufficient. Also not going to get into explicit `ValueError` type checking for every single argument - supported input types are specified in the type annotations and docstrings of every function and should be followed to avoid unintended behaviors.

Reply:
Also, might as well copy that same one-liner logic (assuming it works) to `inspect_all` as well. And maybe `configure_checks`?

Reply:
the one-liner might fail for string inputs with any typo, right? What's wrong with checking for it?

> Also, might as well copy that same one-liner logic (assuming it works) to inspect_all as well. And maybe configure_checks?

sure!

Reply:
>the one-liner might fail for string inputs with any typo, right? What's wrong with checking for it?

The error message from the key error will be enough of an error message in that respect. Not our responsibility to be a spell-checker, which is actually the original reasoning for not even supporting string inputs like this to begin with.

Reply:
Oh, also, be sure to update the annotation type to `Optional[Union[str, Importance]]`